src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
"Actually, don't we need to process removal of diagnostics for different document in same workspace, i.e. even if we get DiagnosticsRemoved when ourDocument.Id != e.DocumentId. Shouldn't we remove the diagnostics as long as the document belongs to our workspace?"

the map it has for diagnostics implicitly use documentId to distinguish diagnostics per document, so it won't be removed even if we call remove.